piq/dss.py

```diff
@@ -162,17 +162,19 @@ def _subband_similarity(x: torch.Tensor, y: torch.Tensor, first_term: bool,
     return similarity
 
 
-def _dct_matrix(size: int) -> torch.Tensor:
+def _dct_matrix(size: int, device: Optional[str] = None, dtype: Optional[type] = None) -> torch.Tensor:
     r""" Computes the matrix coefficients for DCT transform using the following formula:
```

Comment:
Please change `None` defaults to actually wanted default values. What I mean is that even though `device=None` and `dtype=None` do not break the code, they get interpreted by PyTorch as `device='cpu'` and `dtype=torch.float32` implicitly. It is not obvious that `None` casts to these values.

```suggestion
def _dct_matrix(size: int, device: str = 'cpu' dtype: type = torch.float32) -> torch.Tensor:
    r""" Computes the matrix coefficients for DCT transform using the following formula:
```

You might want to remove `from typing import Optional` as well.

Reply:
It was my initial idea to set the default `device` and `dtype` to a common `cpu` and `torch.float` explicitly. However, the standard `PyTorch` functions uses `None` as a default value. 
As we create tensors (filters or meshgrid), I believe that following the PyTorch way of creation makes more sense. In this case, the functionality of our functions `_dct_matrix` (or any filter) matches the official way `torch.zeros()` or `torch.arange()`.
If a user created a tensor with standard pytorch functions, they will expect similar behaviour from our approach.

In case of the use for library purposes, every call of the implemented functions comes with explicit `device` and `dtype` parameters.

Reply:
I see your point @denproc. My greatest fear though is that sometime in the future PyTorch guys may decide, say, that `torch.float16` is so wide spread and efficient for ML/DL that it should become a new default. In this case, our metrics will silently break without us even knowing about it. Do you also see this possibility?

Reply:
In general, this case is covered as our implementation explicitly specifies the `device` and `dtype` with every call. 

An advanced user could use `utils` functions directly for their own purposes. In this case, the change of default behaviour may hit. However, our code will still follow the PyTorch practice, i.e. delegating potential hate-wave to the big player.

Reply:
Fair enough, merging :)